retrieval/scrape.go

```diff
@@ -807,8 +879,11 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	ref, ok := sl.refCache[s]
+	// Suffix s with the invalid \xff unicode rune to avoid collisions
+	// with scraped metrics.
+	s2 := s + "\xff"
```

Comment:
> This line is about caching correctly in the non-simple case.

With "non-simple case", do you mean the target exposing `up` with extra labels? How would those currently collide, we're tracking everything by a series' entire labelset already anyways, no?